src/NUnitFramework/framework/Internal/Execution/ParallelWorkItemDispatcher.cs

```diff
@@ -287,6 +287,12 @@ private void RestoreQueues()
             // Keep lock until we can remove for both methods
             lock (_queueLock)
             {
+                if (_isolationLevel <= 0)
+                {
+                    log.Debug("ParallelWorkItemDispatcher.RestoreQueues(): Isolation level <= 0");
```

Comment:
I think the message should indicate we are ignoring the call.

Reply:
I also wonder if the Guard statement at the start of the method is really appropriate. It's also subject to a race condition and could occur in situations where we really just want to ignore the last call to get in.

Also, maybe the method should be renamed as `TryRestoreQueues` or similar.

Reply:
I agree. I don't think the Guard is necessary. If we run into this scenario, then we will log a Debug message and continue on, which will allow the test run to complete gracefully.